specs/electra/beacon-chain.md

```diff
@@ -640,9 +652,14 @@ def queue_excess_active_balance(state: BeaconState, index: ValidatorIndex) -> No
     if balance > MIN_ACTIVATION_BALANCE:
         excess_balance = balance - MIN_ACTIVATION_BALANCE
         state.balances[index] = MIN_ACTIVATION_BALANCE
-        state.pending_balance_deposits.append(
-            PendingBalanceDeposit(index=index, amount=excess_balance)
-        )
+        validator = state.validators[index]
+        state.pending_deposits.append(PendingDeposit(
+            pubkey=validator.pubkey,
+            withdrawal_credentials=validator.withdrawal_credentials,
+            amount=excess_balance,
+            signature=bls.G2_POINT_AT_INFINITY,
+            slot=GENESIS_SLOT,
```

Comment:
What is the consequence if I set this to `state.slot` instead?

Reply:
Then the excess balance will wait for when `state.slot` is finalized before being applied even in the case when there is enough activation churn to let the balance in. The `GENESIS_SLOT` is used to bypass that finality check as that slot is always finalized